AGENTS.md

```diff
@@ -254,4 +254,173 @@ The compiler is designed for fast feedback loops and scales to large codebases:
 - Remember Lambda IR is the core optimization layer
 - All `lam_*.ml` files process this representation
 - Use `lam_print.ml` for debugging lambda expressions
-- Test both with and without optimization passes
\ No newline at end of file
+- Test both with and without optimization passes
+
+## Working on the Build System
```

Comment:
Please do ready and point out anything missing or slight off.

Reply:
We can do this in future PRs.
Not against the idea, but want to get this in first.